northstar/src/runtime/process/trampoline.rs

```diff
@@ -0,0 +1,26 @@
+use super::{init::Init, Checkpoint};
+use crate::runtime::pipe::Channel;
+use nix::{sched, unistd};
+use std::process::exit;
+
+pub(super) fn trampoline(init: Init, mut child_channel: Channel, checkpoint_init: Checkpoint) -> ! {
+    // Create pid namespace
+    sched::unshare(sched::CloneFlags::CLONE_NEWPID).expect("Failed to create pid namespace");
+
+    // Fork the init process
+    match unsafe { unistd::fork() }.expect("Failed to fork init") {
+        unistd::ForkResult::Parent { child } => {
+            // Send the pid of init to the runtime and exit
+            let pid = child.as_raw() as i32;
+            child_channel.send(&pid).expect("Failed to send init pid");
+            exit(0);
```

Comment:
I think that this causes the `init` process to be an orphan. The `trampolin` process should wait on its exit status.

Reply:
Check `Launcher::start` which does an `prctl(PR_SET_CHILD_SUBREAPER)` to reap all childs that become orphans in this process tree. `init` is a child of the runtime after `trampoline` exited.

Reply:
oh that's cool :)